Develop

PandaGirll opened the pull request:
В данном Pull Request добавлены автотесты для проверки функциональности веб-приложения  «Stellar Burgers». Тесты написаны с использованием фреймворка `pytest` и библиотеки `Selenium WebDriver`. Включены следующие тесты:

1. Регистрация:
   - Успешная регистрация.
   - Проверка ошибки для некорректного пароля.

2. Вход:
   - Вход по кнопке «Войти в аккаунт» на главной.
   - Вход через кнопку «Личный кабинет».
   - Вход через кнопку в форме регистрации.
   - Вход через кнопку в форме восстановления пароля.

3. Переход в личный кабинет:
   - Проверка перехода по клику на «Личный кабинет».

4. Переход из личного кабинета в конструктор:
   - Проверка перехода по клику на «Конструктор».
   - Проверка перехода по клику на логотип Stellar Burgers.

5. Выход из аккаунта:
   - Проверка выхода по кнопке «Выйти» в личном кабинете.

6. Раздел «Конструктор»:
   - Проверка, что работают переходы к разделам: «Булки», «Соусы», «Начинки».

Также добавлен генератор учетных записей для создания тестовых данных.

PandaGirll committed:
Добавлены фикстуры, локаторы, README, генераторы учёток, тесты регистрации и логина.

PandaGirll committed:
Тест разлогина, косметические правки по комментариям

PandaGirll committed:
Тест перехода в личный кабинет, удаление лишних импортов

PandaGirll committed:
Тесты перехода из личного кабинета в конструктор

PandaGirll committed:
Тесты переходов к разделам конструктора. Добавлены локаторы для разделов конструктора.

PandaGirll committed:
Избавились от папки .idea

PandaGirll committed:
Дополнен README.md

PandaGirll committed:
Delete .idea directory

PandaGirll committed:
Update README.md

PandaGirll committed:
Оптимизация внешнего вида кода

irisqul commented on conftest.py:
```diff
@@ -0,0 +1,25 @@
+
+
+# фикстура драйвера
+@pytest.fixture()
```
Отлично: написана фикстура

irisqul commented on data.py:
```diff
@@ -0,0 +1,51 @@
+# Ссылки на страницы
+class TestLinks:
+    main_page_link = "https://stellarburgers.nomoreparties.site/"
+    login_page_link = "https://stellarburgers.nomoreparties.site/login"
```
Можно лучше: Код станет более поддерживаемым, если вынести адрес сервера в отдельную переменную, а в переменных эндпоинтов оставить только их уникальный путь. Таким образом, если нужно будет тестировать не продакшн или если поменяется url будет достаточно обновить одну переменную

irisqul commented on data.py:
```diff
@@ -0,0 +1,51 @@
+        while True:
+            if is_registration:
+                # Предполагаем, что страница регистрации уже открыта
+                driver.find_element(*TestLocators.NAME_LOCATOR).send_keys("Тест")
```
Можно лучше: Это вероятно уже хелпер, а не данные, лучше разместить не в  data

irisqul commented on data.py:
```diff
@@ -0,0 +1,51 @@
+                        EC.visibility_of_element_located(TestLocators.ERROR_MESSAGE_LOCATOR)
+                    )
+                    # Если ошибка появилась, генерируем новый email
+                    email = Generator.generate_email()
```
Можно лучше: Это интересное решение, но оно очень громоздкое и не отвечает принципам программирования, так как метод решает слишком много задач. В следующих спринтах будет предложен более элегантный способ создания тестовой архитектуры, пока - скажу что ветвление (особенно скрытое в хэлперах) - это не удачная идея, мы хотим получить линейные предсказуемые шаги. и нужно разбить код на разные методы - залогин, регистрация, проверки статуса (залогинены ли мы, зарегистрированы ли).

irisqul commented on tests/test_constructor_section.py:
```diff
@@ -0,0 +1,95 @@
+    def test_navigation_to_buns(self, driver, test_email, test_password):
+        # Переход на главную страницу
+        driver.get(TestLinks.main_page_link)
+
```
Можно лучше: Здесь и далее. Проследи за соблюдением стиля PEP8 в твоем коде, между функциями нужно делать отступ в 2 строки, внутри функций - отступы не делаются

PandaGirll replied:
> Согласно PEP 8, определения функций и классов верхнего уровня должны быть окружены двумя пустыми линиями.
> 
> Определения методов внутри класса должны быть окружены одной пустой линией.

Я пыталась делать между функциями отступ в две линии, но flake8 и autopep8 на них ругаются и убирают

irisqul commented on tests/test_constructor_section.py:
```diff
@@ -0,0 +1,95 @@
+        driver.execute_script("arguments[0].scrollIntoView(true);", buns_section)
+
+        # Проверка, что раздел «Булки» виден в области просмотра
+        assert buns_section.is_displayed(), "Ошибка отображения раздела «Булки»."
```
Нужно исправить: чтобы протестировать переход нужно сначала уйти с вкладки, ведь она дефолтно отображается

irisqul commented on tests/test_constructor_section.py:
```diff
@@ -0,0 +1,95 @@
+        driver.execute_script("arguments[0].scrollIntoView(true);", sauces_section)
+
+        # Проверка, что раздел «Соусы» виден в области просмотра
+        assert sauces_section.is_displayed(), "Ошибка отображения раздела «Соусы»."
```
Нужно исправить: наличие текста у этих элементов не говорит о том, что переход на таб произошел. У элемента при клике добавляется новый класс. Следует проверять его

irisqul commented on tests/test_login.py:
```diff
@@ -0,0 +1,84 @@
+        # Переходим на страницу регистрации
+        driver.get(TestLinks.registration_page_link)
+        # Регистрация, попадаем на страничку логина
+        AuthHelper.authenticate(driver, test_email, test_password, is_registration=True)
```
Можно лучше: в тесте проверки логина вынести регистрацию пользователя в фикстуру

irisqul commented on tests/test_logout.py:
```diff
@@ -0,0 +1,29 @@
+        AuthHelper.authenticate(driver, test_email, test_password, is_registration=True)
+
+        # Логин с зарегистрированными данными, попадаем на главную страницу
+        AuthHelper.authenticate(driver, test_email, test_password)
```
Можно лучше: сделать залогин фикстурой, в этом тесте - это предусловие

PandaGirll committed:
Добавлена отдельная переменная для адреса тестируемого сервера.
Изменены переменные эндпоинтов.

PandaGirll committed:
Регистрация пользователей в тестах логина вынесена в фикстуру
Применён новый метод для логина
Код оптимизирован в соответствии с PEP8

PandaGirll committed:
Исправлен test_logout.py - предусловие (регистрация и логин) вынесено в фикстуру
Мелкие правки test_login.py

PandaGirll committed:
Функция универсальной авторизации из файла data.py вынесена в отдельный файл auth_helper.py и разделена на независимые методы.
Исправлен test_registration.py, старый метод регистрации заменён на новый.
Код приведён к соответствию стандарту PEP8.

PandaGirll committed:
Исправлен test_navigation_to_account.py, старая функция регистрации и авторизации заменена на новую фикстуру.
Код приведён к соответствию стандарту PEP8.

PandaGirll committed:
Исправлен test_navigation_to_constructor.py, старая функция регистрации и авторизации заменена на новую фикстуру.
Код приведён к соответствию стандарту PEP8.

PandaGirll committed:
Заменены Локаторы для тестов конструктора.
Исправлен test_constructor_section.py, в тестах перехода по вкладкам используются новые локаторы класса.

Код приведён к соответствию стандарту PEP8.
